sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -42,7 +42,144 @@ APIs for getting statistics about devices and modules, as well as service statis
 APIs for managing device identities, device twins, and querying devices
 
 ```csharp
-
+public class Devices
+{
+	/// <summary>
+	/// Create or update a device.
+	/// </summary>
+	/// <param name="deviceId">The unique identifier of the device to create.</param>
+	/// <param name="device">The device to create or update.</param>
+	/// <param name="ifMatch">A string representing a weak ETag for this device, as per RFC7232. The update operation is performed
+	/// only if this ETag matches the value maintained by the server, indicating that the device has not been modified since it was last retrieved.
+	/// To force an unconditional update, set If-Match to the wildcard character (*). For create operations, this value must be null.</param>
+	/// <param name="cancellationToken">The cancellation token.</param>
+	/// <returns>The created or updated device.</returns>
+	public virtual Response<DeviceIdentity> CreateOrUpdate(string deviceId, DeviceIdentity device, string ifMatch = null, CancellationToken cancellationToken = default)
+
+	/// <summary>
+	/// List devices. This API does not support paging.
+	/// This is deprecated by the service, but there isn't a 1:1 replacement for it. Query returns twins, not devices.
+	/// </summary>
+	/// <param name="top">The max number of devices to list.</param>
+	/// <param name="cancellationToken">The cancellation token.</param>
+	/// <returns>The list of devices retrieved.</returns>
+	public virtual Response<IReadOnlyList<DeviceIdentity>> List(int? top = null, CancellationToken cancellationToken = default)
```

Comment:
For the life of me, I can't figure out what "top" is an acronym for, but its defined this way in the swagger

Reply:
In query parlance, top meaning it will return a max number of items from the "top" of the query results. So if you have 1000 items, and you specify a top of 50, you'll get items 1 to 50. You can't get items 51-100 because it isn't off the top.